### ISSUE TITLE
Can not set per-dataloader seed generator

### ISSUE THREAD

**Issue by chnyutao** [BirkhoffG/jax-dataloader#50]
Hi,

The [document](https://birkhoffg.github.io/jax-dataloader/examples/random/#method-2-setting-per-dataloader-seed-generators) says that you can set per-dataloader seed generator using JAX keys. But with `jax_dataloader=0.1.3` (which seems to be the latest) I can not actually do this -- please see example below.

```python
>>> import jax.numpy as jnp
>>> import jax.random as jr
>>> import jax_dataloader as jdl
>>> xs = jnp.arange(100.)
>>> ds = jdl.ArrayDataset(xs, asnumpy=False)

>>> next(iter(jdl.DataLoader(ds, backend='jax', shuffle=True, generator=jr.key(0))))
(array([83.], dtype=float32),)

>>> next(iter(jdl.DataLoader(ds, backend='jax', shuffle=True, generator=jr.key(1))))
(array([83.], dtype=float32),)

>>> next(iter(jdl.DataLoader(ds, backend='jax', shuffle=True, generator=jr.key(2))))
(array([83.], dtype=float32),)

>>> next(iter(jdl.DataLoader(ds, backend='jax', shuffle=True, generator=jr.key(42))))
(array([83.], dtype=float32),)

>>> jdl.manual_seed(0)
>>> next(iter(jdl.DataLoader(ds, backend='jax', shuffle=True)))
(array([98.], dtype=float32),)

>>> jdl.manual_seed(1)
>>> next(iter(jdl.DataLoader(ds, backend='jax', shuffle=True)))
(array([94.], dtype=float32),)

>>> jdl.manual_seed(2)
>>> next(iter(jdl.DataLoader(ds, backend='jax', shuffle=True)))
(array([74.], dtype=float32),)

>>> jdl.manual_seed(42)
>>> next(iter(jdl.DataLoader(ds, backend='jax', shuffle=True)))
(array([83.], dtype=float32),)
```

Looking into the source code, it seems at least the [`DataLoaderJAX`](https://github.com/BirkhoffG/jax-dataloader/blob/566beb9733765731316e1f1c72cde7d51d582c5d/jax_dataloader/loaders/jax.py#L40) backend simply ignore the `generator` argument in `**kwargs`. Would be happy to submit a PR if you intend to fix this :)

**Comment by BirkhoffG** [comment]
Hi @chnyutao thanks for raising this issue. Yes, this is a bug. Feel free to submit a PR.

Note that we are using nbdev for development/document/test. See: https://nbdev.fast.ai/tutorials/tutorial.html#make-your-first-edit

**Comment by BirkhoffG** [comment]
@chnyutao I revisited this issue, and I cannot reproduce your error in the latest mater branch. This is supposed to fixed by #45 

Pls pull the latest master code and try again. If you still face this issue even if you have the latest code, please re-open this issue.

Will publish a new version shortly.

**Comment by chnyutao** [comment]
Thanks a ton. This seems to be fixed.
